lambda/update/correction.go

```diff
@@ -72,9 +98,59 @@ func validateCorrection(changes []shared.Change, lpa *shared.Lpa) (Correction, [
 		Field("/donor/dateOfBirth", &data.DonorDob, parse.Validate(func() []shared.FieldError {
 			return validate.Date("", data.DonorDob)
 		}), parse.Optional()).
-		Field("/signedAt", &data.LPASignedAt, parse.Validate(func() []shared.FieldError {
+		Field(signedAt, &data.LPASignedAt, parse.Validate(func() []shared.FieldError {
 			return validate.Time("", data.LPASignedAt)
 		}), parse.Optional()).
+		Prefix("/attorneys", func(p *parse.Parser) []shared.FieldError {
+			return p.
+				Each(func(i int, p *parse.Parser) []shared.FieldError {
+					if data.Index != nil && *data.Index != i {
+						return p.OutOfRange()
+					}
+
+					data.Index = &i
+					data.AttorneyFirstNames = lpa.Attorneys[i].FirstNames
+					data.AttorneyLastName = lpa.Attorneys[i].LastName
+					data.AttorneyDob = lpa.Attorneys[i].DateOfBirth
+					data.AttorneyAddress = lpa.Attorneys[i].Address
+					data.AttorneyEmail = lpa.Attorneys[i].Email
+					data.AttorneyMobile = lpa.Attorneys[i].Mobile
+
+					if lpa.Attorneys[i].SignedAt != nil {
+						data.AttorneySignedAt = *lpa.Attorneys[i].SignedAt
+					}
+
+					return p.
+						Field("/firstNames", &data.AttorneyFirstNames, parse.Validate(func() []shared.FieldError {
+							return validate.Required("", data.AttorneyFirstNames)
+						}), parse.Optional()).
+						Field("/lastName", &data.AttorneyLastName, parse.Validate(func() []shared.FieldError {
+							return validate.Required("", data.AttorneyLastName)
+						}), parse.Optional()).
+						Field("/dateOfBirth", &data.AttorneyDob, parse.Validate(func() []shared.FieldError {
+							return validate.Date("", data.AttorneyDob)
+						}), parse.Optional()).
+						Field("/email", &data.AttorneyEmail, parse.Optional()).
+						Field("/mobile", &data.AttorneyMobile, parse.Optional()).
+						Prefix("/address", func(p *parse.Parser) []shared.FieldError {
+							return p.
+								Field("/line1", &data.AttorneyAddress.Line1, parse.Optional()).
+								Field("/line2", &data.AttorneyAddress.Line2, parse.Optional()).
+								Field("/line3", &data.AttorneyAddress.Line3, parse.Optional()).
+								Field("/town", &data.AttorneyAddress.Town, parse.Optional()).
+								Field("/postcode", &data.AttorneyAddress.Postcode, parse.Optional()).
+								Field("/country", &data.AttorneyAddress.Country, parse.Validate(func() []shared.FieldError {
+									return validate.Country("", data.AttorneyAddress.Country)
+								}), parse.Optional()).
```

Comment:
you could extract the address bit like:

```go
func validateAddress(address *shared.Address) func(p *parse.Parser) []shared.FieldError {
	return func(p *parse.Parser) []shared.FieldError {
		return p.
			Field("/line1", &address.Line1, parse.Optional()).
			Field("/line2", &address.Line2, parse.Optional()).
			Field("/line3", &address.Line3, parse.Optional()).
			Field("/town", &address.Town, parse.Optional()).
			Field("/postcode", &address.Postcode, parse.Optional()).
			Field("/country", &address.Country, parse.Validate(func() []shared.FieldError {
				return validate.Country("", address.Country)
			}), parse.Optional()).
			Consumed()
	}
}

// validateAddress(&data.DonorAddress)
// validateAddress(&data.AttorneyAddress)
```

Reply:
you could also use a similar technique to extract the `/donor` and `/attorney` handling if you wanted